src/utils/ipfs.ts

```diff
@@ -10,12 +10,21 @@ import { THE_BADGE_BACKEND_URL } from '@utils/constants'
  * @param {string} hash - The IPFS hash representing the content to retrieve.
  */
 // eslint-disable-next-line @typescript-eslint/ban-types
-export async function getFromIPFS<T, X = {}>(hash?: string): Promise<unknown> {
-  if (!hash) return
-  const cleanedHash = hash.replace(/^ipfs?:\/\//, '')
-  if (!cleanedHash) return
+export async function getFromIPFS<T, X = {}>(hash: string): Promise<BackendResponse<{ content: T } & X>> {
+  const errorResponse: BackendResponse<{ content: T } & X> = {
+    error: true,
+    statusCode: 404,
+    result: null,
+  }
 
-  return await axios.get<BackendResponse<{ content: T } & X>>(`${THE_BADGE_BACKEND_URL}/api/ipfs/${cleanedHash}`)
+  if (!hash) return errorResponse
+  const cleanedHash = hash.replace(/^ipfs?:\/\//, '').replace(/^ipfs?\//, '')
```

Comment:
You can move this a helper, as we have it on the front end, since it will be used a lot

Reply:
done